flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentControllerTest.java

```diff
@@ -274,6 +279,15 @@ public void verifyInProgressDeploymentWithCrashLoopBackoff() throws Exception {
         validatingResponseProvider.assertValidated();
     }
 
+    @ParameterizedTest
+    @EnumSource(FlinkVersion.class)
+    public void cleanUpDeploymentWithError(FlinkVersion flinkVersion)
+            throws JsonProcessingException {
+        FlinkDeployment flinkDeployment = TestUtils.buildApplicationCluster();
+        val deleteControl = testController.cleanup(flinkDeployment, context);
```

Comment:
I think you accidentally used lombok `val` instead of `var`

Reply:
Can I use val in the tests? I prefer to use final instead of a mutable variable.

Reply:
We generally don’t use this and also avoid final variables in these cases as it doesn’t affect anything only adds boilerplate on these cases.

also we should restrict Lombok usage where it adds significant tangible value

Reply:
final variables inside methods are a bit of a controversy in Java. I would say the flink best practice is to only use this in performance critical low level code

Reply:
ok